spikeinterface/core/basesorting.py

```diff
@@ -15,6 +14,7 @@ def __init__(self, sampling_frequency: float, unit_ids: List):
         BaseExtractor.__init__(self, unit_ids)
         self._sampling_frequency = sampling_frequency
         self._sorting_segments: List[BaseSortingSegment] = []
+        self._recording = None
```

Comment:
Maybe I would choose a more explicit naming.
And you should put some comment in the code like
"# this is to handle times vector with a link..."